specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -12,821 +12,329 @@
-export class WbView extends Backbone.View {
-  public readonly data: RA<RA<string | null>>;
-
-  public readonly mappings: WbMapping | undefined = undefined;
-
-  // eslint-disable-next-line functional/prefer-readonly-type
-  public hotIsReady: boolean = false;
-
-  // eslint-disable-next-line functional/prefer-readonly-type
-  public hotCommentsContainer: HTMLElement | undefined = undefined;
-
-  // eslint-disable-next-line functional/prefer-readonly-type
-  public undoRedoIsHandled: boolean = false;
-
-  private readonly dataSetMeta: DataSetNameView;
-
-  public readonly wbUtils: WbUtils;
-
-  public readonly cells: WbCellMeta;
-
-  public readonly validation: WbValidation;
-
-  public readonly actions: WbActions;
-
-  public readonly disambiguation: Disambiguation;
-
-  // eslint-disable-next-line functional/prefer-readonly-type
-  public hot: Handsontable | undefined;
-
-  /*
-   * If this.isUploaded, render() will:
-   * Add the "Uploaded" label next to DS Name
-   * Disable cell editing
-   * Disable adding/removing rows
-   * Still allow column sort
-   * Still allow column move
-   *
-   */
-  public readonly isUploaded: boolean;
-
-  // Disallow all editing and some tools while this dialog is open
-  // eslint-disable-next-line functional/prefer-readonly-type
-  public uploadedView: (() => void) | undefined = undefined;
-
-  // Disallow all editing and all tools while this dialog is open
-  // eslint-disable-next-line functional/prefer-readonly-type
-  public coordinateConverterView: (() => void) | undefined = undefined;
-
-  public readonly handleResize: () => void;
-
-  public readonly throttleRate: number;
-
-  // Constructors & Renderers
-  public constructor(
-    public readonly dataset: Dataset,
-    // eslint-disable-next-line functional/prefer-readonly-type
-    public refreshInitiatedBy: WbStatus | undefined,
-    // eslint-disable-next-line functional/prefer-readonly-type
-    public refreshInitiatorAborted: boolean,
-    public readonly element: HTMLElement,
-    public readonly options: {
-      readonly onSetUnloadProtect: (unloadProtect: boolean) => void;
-      readonly onDeleted: () => void;
-      readonly onDeletedConfirmation: () => void;
-      readonly display: (
-        jsx: JSX.Element,
-        element?: HTMLElement,
-        destructor?: () => void
-      ) => () => void;
-    }
-  ) {
-    super({
-      el: element,
-      tagName: 'section',
-      events: {
-        'click .wb-upload': 'upload',
-        'click .wb-validate': 'upload',
-        'click .wb-data-check': 'toggleDataCheck',
-        'click .wb-show-plan': 'showPlan',
-        'click .wb-revert': 'revertChanges',
-        'click .wb-save': 'save',
-        'click .wb-export-data-set': 'export',
-        'click .wb-change-data-set-owner': 'changeOwner',
-
-        'click .wb-show-upload-view': 'displayUploadedView',
-        'click .wb-unupload': 'unupload',
-      },
-    });
-
-    this.data =
+export type Workbench = {
```

Comment:
could you please add a `readonly ` modifier in front of each of these properties like ESLint is suggesting:

![Screenshot 2024-04-16 at 18 14 59](https://github.com/specify/specify7/assets/40512816/84b5529c-77c6-458f-8150-e01648947c93)

that would prevent you from accidentally changing any of these properties (and thus introducing hard to catch bugs)

unless you explicitly want some of these to be readonly - in which you should explicitly leave those few without a readonly modifier - additional benefit of such approach is that it would clearly show anyone reading the code what of these properties are going or not going to be modified by the code

Reply:
Will do. I think there's something wrong with my ESLint extension config as I don't get any of these suggestions.

Reply:
> I think there's something wrong with my ESLint extension config as I don't get any of these suggestions.

makes sense. The config needs to be updated to match the newest eslint version and updated plugins - something I want to do as soon as this 7.9.4 is out